libraries/chain/protocol/market.cpp

```diff
@@ -43,6 +43,17 @@ void call_order_update_operation::validate()const
    FC_ASSERT( fee.amount >= 0 );
    FC_ASSERT( delta_collateral.asset_id != delta_debt.asset_id );
    FC_ASSERT( delta_collateral.amount != 0 || delta_debt.amount != 0 );
+
+   FC_ASSERT( extensions.size() <= 1, "At most one object is allowed in extensions of call_order_update_operation" );
+   if( extensions.size() > 0 )
+   {
+      const auto& options = extensions.front().value;
+      size_t count = 0; // use a count variable for future expansions
```

Comment:
I wouldn't make assumptions about the logic of future extensions. Just check that tcr is valid right now, instead of counting stuff. (See account extensions for example.)

Reply:
IMHO nonempty is basic requirement for `extensions` field. Additional validation may or may not apply when adding new fields (no more validation this time since TCR is optional). Btw there are more fields to be added as discussed in https://github.com/bitshares/bsips/issues/51, rather than counting after added them, I decided to count from now on.

Reply:
In regards to account extensions, since it's not using a container which can jsonify to an array, but replaced `flat_set<static_variant<X>>` with `extension<Y>` directly (thus have broken JSON compatibility), it should be allowed if the extensions field is empty. So there are individual validations about the fields inside.

I used a vector here, so doesn't make sense to allow an empty object inside. The validation code here is for different things.

Reply:
Even when more fields are added, the actual count is irrelevant, as long as it is > 0.
If we remove the vector wrapper, count==0 will have to be accepted, which makes the whole check here useless.

Reply:
Removed `vector`, so removed validation.